Userland/Libraries/LibCore/LocateDb.cpp

```diff
@@ -32,12 +58,86 @@ void LocateDb::write_directory(DirectoryInfo& dir_info)
     fwrite(dir_chunk.to_string().characters(), sizeof(char), dir_chunk.to_string().length(), file_handle);
 }
 
+bool LocateDb::verify_header()
+{
+    char buffer[BUFSIZ];
+    fgets(buffer, BUFSIZ, file_handle);
+    if (String(buffer) == "locatedb|1\n")
+        return true;
+    return false;
+}
 
-NonnullOwnPtr<DirectoryInfo> LocateDb::get_next_directory()
+LocateDb::RowType LocateDb::identify_row(char* buffer)
 {
-    VERIFY(byte_offset >= sizeof("locatedb|1"));
+    RowType found_type;
+    switch (RowType(buffer[0] - '0')) {
+    case (RowType::dir_start):
+        found_type = RowType::dir_start;
+        break;
+    case (RowType::dir_end):
+        found_type = RowType::dir_end;
+        break;
+    case (RowType::dir):
+        found_type = RowType::dir;
+        break;
+    case (RowType::file):
+        found_type = RowType::file;
+        break;
+    default:
+        dbgln("Buffer length: %zu\n", String(buffer).length());
+        dbgln("Buffer on fail: %s\n", buffer);
```

Comment:
This isn't going to work, `dbgln()`'s placeholders are braces (e.g. `"Buffer on fail: {}"`), and it will automatically append a newline, no need for `\n`.